cudamapper/src/index_gpu.cuh

```diff
@@ -484,27 +491,28 @@ void filter_out_most_common_representations(const double filtering_parameter,
-                                                                                         unique_representation_index_after_filtering_d.data().get(),
-                                                                                         representations_d.data().get(),
-                                                                                         read_ids_d.data().get(),
-                                                                                         positions_in_reads_d.data().get(),
-                                                                                         directions_of_representations_d.data().get(),
-                                                                                         representations_after_compression_d.data().get(),
-                                                                                         read_ids_after_compression_d.data().get(),
-                                                                                         positions_in_reads_after_compression_d.data().get(),
-                                                                                         directions_of_representations_after_compression_d.data().get());
+                                                                                         number_of_sketch_elements_with_each_representation_d.data(),
+                                                                                         first_occurrence_of_representations_d.data(),
+                                                                                         first_occurrence_of_representations_after_compression_d.data(),
+                                                                                         unique_representation_index_after_filtering_d.data(),
+                                                                                         representations_d.data(),
+                                                                                         read_ids_d.data(),
+                                                                                         positions_in_reads_d.data(),
+                                                                                         directions_of_representations_d.data(),
+                                                                                         representations_after_compression_d.data(),
+                                                                                         read_ids_after_compression_d.data(),
+                                                                                         positions_in_reads_after_compression_d.data(),
+                                                                                         directions_of_representations_after_compression_d.data());
 
     // *** swap vectors with the input arrays ***
     std::swap(unique_representations_d, unique_representations_after_compression_d);
```

Comment:
@ahehn-nv - Thanks for the clarification. It is more of a coding style to use namespace explicitly in my coding whenever possible. 
Also, my intention here was to switch `claragenomics::swap` with `std::swap`, I was using `claragenomics::swap` because that was the only way to support the functionality at the beginning when we didn't have move assignment operator implemented. However, as a part of this PR review process, we implemented the move assignment operator which makes `claragenomics::swap` unnecessary. I will remove `claragenomics::swap` from buffer.hpp as well to avoid any future confusion.

Reply:
No, please do not remove `claragenomics::swap`! This should be part of its interface - if you remove it you will always have to call `std::swap` with full namespace qualification.

Reply:
But why do we need `claragenomics::swap` now? Isn't that a redundant functionality.

Reply:
argument-dependent lookup is an integral part of C++. It allows you to overload free functions with more efficient routines for your types/classes (in your namespace).

Reply:
Granted with C++11 and move semantics std::swap has improved a lot and in this case it would do just fine.
But picture the situation for C++03, where std::swap looks like this:
```
void swap(T& a, T& b)
{
  T tmp = a;
  a = b;
  b = tmp;
}
```
If you would use std::swap in this case, it would involve 3 copies.
If you just write swap(a,b) in your code, it will work for the std::* containers, but if your container in his namespace provides its optimized swap function, it will use that - and that may then just be a pointer swap or something.

Reply:
Regarding ADL and swap, IIRC using `std::swap` on `thrust::device_vector` will result exactly in triple copies and allocation and deallocation of device memory for temp. `thrust::swap` simply exchanges pointers (and other internal variables). It would make sense to have the same behavior for our buffer as well

Reply:
Thanks @ahehn-nv and @mimaric. It is now using the `claragenomics::swap` defined for this container which just swaps the internal variables.